src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -189,65 +150,45 @@ public virtual void OnSocketClosed()
             if (_filteredStreamAdapter != null)
             {
                 _filteredStreamAdapter.Abort();
-                _rawSocketInput.IncomingFin();
+                SocketInput.IncomingFin();
                 _readInputTask.ContinueWith((task, state) =>
                 {
                     ((Connection)state)._filterContext.Connection.Dispose();
                     ((Connection)state)._filteredStreamAdapter.Dispose();
-                    ((Connection)state)._rawSocketInput.Dispose();
+                    ((Connection)state).SocketInput.Dispose();
                 }, this);
             }
             else
             {
-                _rawSocketInput.Dispose();
+                SocketInput.Dispose();
             }
 
-            lock (_stateLock)
-            {
-                _connectionState = ConnectionState.SocketClosed;
-
-                if (_socketClosedTcs != null)
-                {
-                    // This is always waited on synchronously, so it's safe to
-                    // call on the libuv thread.
```

Comment:
Might want to keep this comment. Maybe even add that it's only `KestrelThread.Stop()` that uses it. (or `Listener` until your other PR is merged)